### DISCUSSION TITLE
Use of 'version' configuration attribute with wildfly:start

### DISCUSSION THREAD

**Discussion by rachmatowicz** [wildfly/wildfly-maven-plugin#345] [Q&A]
I'm trying to use the wildfly-maven-plugin to be able to start a Wildfly instance using the 'mvn wildfly:start' command.
My plugin configuration looks like this:

>             <plugin>
>                <groupId>org.wildfly.plugins</groupId>
>                <artifactId>wildfly-maven-plugin</artifactId>
>                <version>4.1.0.Final</version>
>                <configuration>
>                    <filename>${project.build.finalName}.war/</filename>
>                    <skip>false</skip>
>                    <version>26.0.0.Final</version>
>                </configuration>
>            </plugin>
I would like to start an instance of Wildfly 26.0.0.Final.
When I run the maven command, I get instead an instance of the latest version of Wildfly, 28.0.1.Final.

Two questions:
- am I using an incorrect plugin configuration to do what I want to do?
- which maven repo are the wildfly GAVs downloaded from and is there anyway to override this? (for example, to start a custo instance of Wildfly that I upload to a maven repo) 

Thanks in advance!

**Comment by jamezp** [comment]
Hi @rachmatowicz,
That configuration looks correct to me. FWIW the `filename` should not be required, but should work.

I tested a configuration like this:
```xml
<plugin>
    <groupId>org.wildfly.plugins</groupId>
    <artifactId>wildfly-maven-plugin</artifactId>
    <version>4.1.0.Final</version>
    <configuration>
        <version>26.0.0.Final</version>
    </configuration>
</plugin>
```
And ran `mvn clean wildfly:start` and it worked for me. Do you have an environment variable of `JBOSS_HOME` set or did you try running `mvn clean` first? It could be you've already got a server in your `target/server` directory which means a new server will not be provisioned.

It downloads the artifacts via normal Maven artifact downloads. Meaning if the artifacts are already available locally, nothing will be downloaded. If they are not available locally, they will be downloaded from the repository you have Maven configured to download from.

If you were not aware, you can override your local repository from the command line with something like `-Dmaven.repo.local=$HOME/tmp/m2/repository`. I do this quite frequently so I don't have to blow away by default repository :)

**Reply by rachmatowicz** [reply]
Hi @jamezp 
I did not have a JBOSS_HOME environment variable set, but I did have a target/server directory populated with a 28.0.1.Final version left over from previous tests. Running mvn clean fixed things up as you eplained.
Thank you for your helpful explaination!
Richard
